sentry/src/main/java/io/sentry/IMetricsAggregator.java

```diff
@@ -0,0 +1,126 @@
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface IMetricsAggregator extends Closeable {
+
+  /**
+   * Emits a Counter metric
+   *
+   * @param key A unique key identifying the metric
+   * @param value The value to be added
+   * @param unit An optional unit, see {@link MeasurementUnit}
+   * @param tags Optional Tags to associate with the metric
+   * @param timestampMs The time when the metric was emitted. Defaults to the time at which the
+   *     metric is emitted, if no value is provided.
+   * @param stackLevel Optional number of stacks levels to ignore when determining the code location
+   */
+  void increment(
+      final @NotNull String key,
+      final double value,
+      final @Nullable MeasurementUnit unit,
+      final @Nullable Map<String, String> tags,
+      final long timestampMs,
+      final int stackLevel);
+
+  /**
+   * Emits a Gauge metric
+   *
+   * @param key A unique key identifying the metric
+   * @param value The value to be added
+   * @param unit An optional unit, see {@link MeasurementUnit}
+   * @param tags Optional Tags to associate with the metric
+   * @param timestampMs The time when the metric was emitted. Defaults to the time at which the
+   *     metric is emitted, if no value is provided.
+   * @param stackLevel Optional number of stacks levels to ignore when determining the code location
+   */
+  void gauge(
+      final @NotNull String key,
+      final double value,
+      final @Nullable MeasurementUnit unit,
+      final @Nullable Map<String, String> tags,
+      final long timestampMs,
+      final int stackLevel);
+
+  /**
+   * Emits a Distribution metric
+   *
+   * @param key A unique key identifying the metric
+   * @param value The value to be added
+   * @param unit An optional unit, see {@link MeasurementUnit}
+   * @param tags Optional Tags to associate with the metric
+   * @param timestampMs The time when the metric was emitted. Defaults to the time at which the
+   *     metric is emitted, if no value is provided.
+   * @param stackLevel Optional number of stacks levels to ignore when determining the code location
+   */
+  void distribution(
+      final @NotNull String key,
+      final double value,
+      final @Nullable MeasurementUnit unit,
+      final @Nullable Map<String, String> tags,
+      final long timestampMs,
+      final int stackLevel);
+
+  /**
+   * Emits a Set metric
+   *
+   * @param key A unique key identifying the metric
+   * @param value The value to be added
+   * @param unit An optional unit, see {@link MeasurementUnit}
+   * @param tags Optional Tags to associate with the metric
+   * @param timestampMs The time when the metric was emitted. Defaults to the time at which the
+   *     metric is emitted, if no value is provided.
+   * @param stackLevel Optional number of stacks levels to ignore when determining the code location
+   */
+  void set(
```

Comment:
is the set with `int` wanted?
Again, no mention of types in the develop docs. The [only example](https://develop.sentry.dev/delightful-developer-metrics/sending-metrics-sdk/) suggests Strings.
This applies to other methods as well (int or doubles?) but i guess double is fine there

Reply:
Yes, it's either an `int` or a cr32 encoded string: https://develop.sentry.dev/sdk/metrics/#sets.
IMHO we could also deviate here and make it Java friendly: Accepting `Object` and using `hashCode()` under the hood